### PR TITLE
fix(page): mock selection of multiline text on create/edit link

### PR DESCRIPTION
**Pull request by AyushAgrawal-A2**
Closes: #5014 

Changes:
* Mulitple mock selection divs, derived using `getClientRects`, for representing text selection accurately.
* Check for `isValidUrl` in the `_onConfirm` method before `normalizeUrl`. Previously, an invalid URL was getting accepted on pressing the Enter key as this check was after normalizeUrl.

https://github.com/toeverything/blocksuite/assets/54364088/2a39ac1d-923e-4f7c-ba00-1b0fc3a49545

### CODE REVIEW

File: packages/blocks/src/_common/components/rich-text/virgo/nodes/link-node/link-popup/link-popup.ts [hunk 1]
```diff
@@ -87,12 +87,19 @@ export class LinkPopup extends WithDisposable(LitElement) {
...
+      Object.values(domRects).forEach(domRect => {
+        const mockSelection = document.createElement('div');
+        mockSelection.classList.add('mock-selection');
+        mockSelection.style.left = `${domRect.left}px`;
+        mockSelection.style.top = `${domRect.top}px`;
+        mockSelection.style.width = `${domRect.width}px`;
+        mockSelection.style.height = `${domRect.height}px`;
+
+        assertExists(this.mockSelectionContainer);
+        this.mockSelectionContainer.appendChild(mockSelection);
```
Comment (doodlewind) [inline packages/blocks/src/_common/components/rich-text/virgo/nodes/link-node/link-popup/link-popup.ts:101]: Should the container be cleaned up before appending?

Comment (AyushAgrawal-A2) [reply packages/blocks/src/_common/components/rich-text/virgo/nodes/link-node/link-popup/link-popup.ts:101]: The link-popup is a fresh element added to dom every time create / edit link called, with an empty container.
It does not update during its lifecycle. Clearing it seems not nessesary in the current implementation.

**Comment by AyushAgrawal-A2** [comment]
Hi @doodlewind,
While solving this issue i noticed that while creating a new link, if the selection has some text with different text attributes multiple link elements are created :-

https://github.com/toeverything/blocksuite/assets/54364088/ff86a1cb-8759-40f2-8c2e-1953e2aa75ac


Shouldn't it be a single link with all text attributes removed ?

**Comment by doodlewind** [comment]
@AyushAgrawal-A2 thanks for the finding, removing all attributes here makes sense - new PR welcome :)

**Review by doodlewind — APPROVED** [review]

**Pull request merged**